source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -122,18 +122,6 @@ OslShaderGenerator::OslShaderGenerator()
     registerImplementation("IM_swizzle_vector4_vector2_sx_osl", Swizzle::create);
     registerImplementation("IM_swizzle_vector4_vector3_sx_osl", Swizzle::create);
     registerImplementation("IM_swizzle_vector4_vector4_sx_osl", Swizzle::create);
-
```

Comment:
Remove old remapping since 2 and 4 structs are now supported.

Reply:
Sure. I can do that. We currently always use an ArnoldShaderGenerator for testing. I assume a  generic OslShaderGenerator can't be instantiated so it will produce what it did before. As it is now the output surface shader does basically what this remapping does as it's output is always a color3. 

How about I revert this in the base class but add in an option of whether to remap or not. The option is set to true in the derived Arnold class. Then the option is still there in case anyone else's  compiler does not handle the struct output.